SharkCage/CageManager/CageManager.cpp

```diff
@@ -54,15 +83,16 @@ int main()
 		&CageManager::StartCage,
 		cage_manager,
 		security_attributes.value(),
-		cage_data
+		cage_data,
+		group_name
 	);
 
 	desktop_thread.join();
-
+	tokenLib::deleteLocalGroup(static_cast<LPWSTR>(const_cast<wchar_t*>((group_name.c_str()))));
 	return 0;
 }
 
-void CageManager::StartCage(SECURITY_ATTRIBUTES security_attributes, const CageData &cage_data)
+void CageManager::StartCage(SECURITY_ATTRIBUTES security_attributes, const CageData &cage_data, const std::wstring group_name)
```

Comment:
could ne `group_name` be added to the `CageData` struct? or did you keep it seperate on purpose? also, can you please make this a const reference so the string does not have to be copied?

Reply:
I made this decision because of two reasons:

1) **What CageData represents semantically**: `CageData` is a structure that is passed through several layers of the system basically unchanged and it holds only non-temporary data related to caged application directly. The group is a) temporary and b) random (thus having no realtionship with the actual caged app)

2) **Inicialization**: _(this one is completely practical)_ `CageData` is incialized in ´CageManager.cpp´ directly from the message string: `CageData cage_data = { message_data };` I assume I would need to rewrite that if I add another member to `CageData`. And I did not want to touch it.